### ISSUE TITLE
Source Code

### ISSUE THREAD

**Issue by no-clue-what-to-do** [iremzog/RamanDenoising#1]
Is there any way you could put up the code that makes the models? I'd like to be able to play around with the architecture and see exactly how the models were implemented

Thanks!

**Comment by iremzog** [comment]
Hello, you may find the code snipet for the FCED and the custom loss function below. All implemented in Keras and Tensorflow. Hope you find these helpful! :)

```
def FCED(input_size, loss = 'mse'):

    input = Input(shape=input_size)

    # Encoder Part
    x = Conv1D(filters = 16, kernel_size = 3, strides = 1, padding = 'same', 
                   activation = 'relu', kernel_initializer = 'glorot_uniform')(input)
    x = Conv1D(filters = 16, kernel_size = 3, strides = 1, padding = 'same', 
                   activation = 'relu', kernel_initializer = 'glorot_uniform')(x)
    x = MaxPooling1D(pool_size= 2, strides=None, padding='valid')(x)

    for f in [32, 64]:
      x = Conv1D(filters = f, kernel_size = 3, strides = 1, padding = 'same', 
                    activation = 'relu', kernel_initializer = 'glorot_uniform')(x)
      x = Conv1D(filters = f, kernel_size = 3, strides = 1, padding = 'same', 
                    activation = 'relu', kernel_initializer = 'glorot_uniform')(x)
      x = MaxPooling1D(pool_size= 2, strides=None, padding='valid')(x)

    # Decoder Part
    for f in [64, 32, 16]:
      x = Conv1DTranspose(filters = f, kernel_size = 3, strides = 1, padding = 'same', 
                    activation = 'relu', kernel_initializer = 'glorot_uniform')(x)
      x = Conv1DTranspose(filters = f, kernel_size = 3, strides = 1, padding = 'same', 
                    activation = 'relu', kernel_initializer = 'glorot_uniform')(x)
      x = UpSampling1D(2)(x)

    decoded = Conv1D(filters = 1, kernel_size = 3, activation=None, padding='same')(x)

    model = Model(inputs = input, outputs = decoded)
    model.compile(optimizer = optimizers.Adam(learning_rate = 1e-4), loss = loss)

    return model
```

```
def custom_loss(alpha, thr_ratio):

  def peak_mse_combined_loss(y_true, y_pred): 

    y_true = tf.cast(y_true, tf.float32)
    y_pred = tf.cast(y_pred, tf.float32)
    mse = tf.keras.losses.mean_squared_error(y_true, y_pred)
    
    thr = tf.reduce_max(y_true, axis=1, keepdims=True)*thr_ratio
    weights = y_true > thr
    weights = tf.cast(weights, tf.float32)
    y_true_peaks = tf.math.multiply(weights, y_true)
    y_pred_peaks = tf.math.multiply(weights, y_pred)

    mse_peak = tf.keras.losses.mean_squared_error(y_true_peaks, y_pred_peaks)

    return mse + mse_peak*alpha

  return peak_mse_combined_loss
```

**Comment by no-clue-what-to-do** [comment]
Thank you so much! This is very helpful, especially the code for the loss!
